include/string_utils.h

```diff
@@ -139,6 +139,47 @@ void reset_str(T *str) noexcept
 	terminate_str_at(str, 0);
 }
 
+// Is the character within the standard ASCII range?
+constexpr bool is_ascii(char c)
+{
+	// The standard ASCII range spans the lower 7 bits
+	// Ref: https://en.wikipedia.org/wiki/ASCII#7-bit_codes
+	constexpr char first_valid = 0b000'0000;
+	constexpr char last_valid  = 0b111'1111;
+	return c >= first_valid && c <= last_valid;
+}
+
+// Is it an ASCII control character?
+constexpr bool is_control_ascii(char c)
+{
+	// ASCII control characters span the bottom 5 bits, plus the last
+	// character. Ref: https://en.wikipedia.org/wiki/ASCII#Control_characters
+	constexpr char bottom_5b_first = 0b0'0000;
+	constexpr char bottom_5b_last  = 0b1'1111;
+	constexpr char last_ascii_val  = 0b111'1111;
+	return (c >= bottom_5b_first && c <= bottom_5b_last) ||
```

Comment:
If  'char' is unsigned, comparison "c >= bottom_5b_first" is going to be always true - some compilers are going to complain.

Reply:
Ahh.. good catch (you've been here before!).
This is where I'll place your CHAR_MIN preprocessor check (hide it away from user-side code).

Reply:
Added